ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/common/GenesisWorldStateProvider.java

```diff
@@ -65,7 +65,7 @@ private static MutableWorldState createGenesisBonsaiWorldState() {
         new BonsaiWorldStateKeyValueStorage(
             new KeyValueStorageProvider(
                 segmentIdentifiers -> new SegmentedInMemoryKeyValueStorage(),
-                new InMemoryKeyValueStorage(),
+                () -> new InMemoryKeyValueStorage(),
```

Comment:
this should use the actual configured preimage store, otherwise we will miss genesis preimages in the preimage store

Reply:
this path does not update state, it is just for initialization.